app/views/handlers/block.py

```diff
@@ -125,16 +125,33 @@ def _set_started_at_metadata(self):
 
     def _get_safe_page_title(self, page_title):
         page_title = self._schema.get_single_string_value(page_title)
-
         return safe_content(page_title)
 
     def _resolve_custom_page_title_vars(self) -> MutableMapping:
-        if list_item_id := self.current_location.list_item_id:
-            list_item_position = (
-                self._questionnaire_store.list_store.list_item_position(
-                    self.current_location.list_name, list_item_id
-                )
-            )
-            return {"list_item_position": list_item_position}
+        list_item_position = self._questionnaire_store.list_store.list_item_position(
+            self.current_location.list_name, self.current_location.list_item_id
+        )
+        return {"list_item_position": list_item_position}
 
-        return {}
+    def _get_content_title(self, transformed_block):
+        content = transformed_block.get("content")
+        if content:
+            return self._get_safe_page_title(content["title"])
+
+    def _set_page_title(self, variant_page_title):
+        self.page_title = variant_page_title
+
+        section_repeating_page_title = (
+            self._schema.get_repeating_page_title_for_section(
+                self._current_location.section_id
+            )
+        )
+        if section_repeating_page_title:
+            self.page_title = f"{self.page_title}: {section_repeating_page_title}"
+
+        if self.page_title and (
+            self._current_location.list_item_id
+            or self.block["type"] == "ListAddQuestion"
+        ):
+            page_title_vars = self._resolve_custom_page_title_vars()
+            self.page_title = self.page_title.format(**page_title_vars)
```

Comment:
- `self.page_title` should be set once at the end of the method
- The method should always be called with a title, so we don't need to check for `page_title` in the second conditional

```suggestion
    def _set_page_title(self, page_title):
        section_repeating_page_title = (
            self._schema.get_repeating_page_title_for_section(
                self._current_location.section_id
            )
        )
        if section_repeating_page_title:
            page_title = f"{page_title}: {section_repeating_page_title}"

        if (
            self._current_location.list_item_id
            or self.block["type"] == "ListAddQuestion"
        ):
            page_title_vars = self._resolve_custom_page_title_vars()
            page_title = page_title.format(**page_title_vars)

        self.page_title = page_title
```